### ISSUE TITLE
[staticcheck,SA4006] false positives

### ISSUE THREAD

**Issue by supercairos** [golangci/golangci-lint#6363]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/docs/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/docs/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

Official binary

### Description of the problem

Using new Go v1.26, the `new(typeB(varTypeA))` escapes staticcheck and yeald `SA4006: this value of offset is never used` 

```
json, err := geojson.NewGeometry(orb.Point{*input.Longitude, *input.Latitude}).MarshalJSON()
if err != nil {
	return nil, fmt.Errorf("failed to marshal geolocation to GeoJSON: %w", err)
}

geomJson = new(string(json)) <--- HERE
```

### Version of golangci-lint

<details>
golangci-lint has version 2.9.0 built with go1.26.0 from 72798d34 on 2026-02-10T23:13:22Z
</details>

### Configuration

<details>

```console
# paste configuration file or CLI flags here
```

</details>

### Go environment

<details>

```console
version: "2"

linters:
  default: all
  disable:
    - unqueryvet
    - depguard
    - nlreturn
    - lll
    - exhaustruct
    - tagalign
    - godot
    - wsl
    - revive
    - bodyclose
    - gochecknoglobals
    - gochecknoinits
    - ireturn
    - forcetypeassert
    - tparallel
    - godoclint
  exclusions:
    rules:
      # disable funlen for all _test.go files
      - path: test_config.go
        linters:
          - mnd
      - path: _test.go
        linters:
          - errchkjson
          - maintidx
          - funlen
          - varnamelen
          - err113
          - errcheck
  settings:
    cyclop:
      max-complexity: 15
    funlen:
      lines: 80
      statements: 40
      ignore-comments: true
    staticcheck:
      # https://staticcheck.dev/docs/configuration/options/#dot_import_whitelist
      # Default: ["github.com/mmcloughlin/avo/build", "github.com/mmcloughlin/avo/operand", "github.com/mmcloughlin/avo/reg"]
      dot-import-whitelist:
        - github.com/go-jet/jet/v2
        - github.com/go-jet/jet/v2/postgres
        - github.com/supercairos/family-website/pkg/jetx
        - github.com/supercairos/family-website/internal/database/.gen/directory/public/table

    tagliatelle:
      # Checks the struct tag name case.
      case:
        rules:
          json: snake
        overrides:
          - pkg: pkg/clients/helloasso
            rules:
              json: camel
    varnamelen:
      ignore-names:
        - err
        - ctx
        - id
        - ids
        - db
        - tx
      ignore-decls:
        - c *gin.Context

    paralleltest:
      ignore-missing-subtests: true

    recvcheck:
      exclusions:
        - "*.Value"
        - "*.Scan"

formatters:
  enable:
    - gci
    - gofmt
    - gofumpt
    - goimports
    - golines

```

</details>

### Verbose output of running

<details>

```console
INFO golangci-lint has version 2.9.0 built with go1.26.0 from 72798d34 on 2026-02-10T23:13:22Z 
INFO [config_reader] Config search paths: [./ /Users/romain/git/FamilyWebsite/backend /Users/romain/git/FamilyWebsite /Users/romain/git /Users/romain /Users /] 
INFO [config_reader] Used config file .golangci.yml 
INFO [config_reader] Module name "github.com/supercairos/family-website" 
INFO [goenv] Read go env for 7.016875ms: map[string]string{"GOCACHE":"/Users/romain/Library/Caches/go-build", "GOROOT":"/usr/local/go"} 
INFO [lintersdb] Active 101 linters: [arangolint asasalint asciicheck bidichk canonicalheader containedctx contextcheck copyloopvar cyclop decorder dogsled dupl dupword durationcheck embeddedstructfieldcheck err113 errcheck errchkjson errname errorlint exhaustive exptostd fatcontext forbidigo funcorder funlen gci ginkgolinter gocheckcompilerdirectives gochecksumtype gocognit goconst gocritic gocyclo godox gofmt gofumpt goheader goimports golines gomoddirectives gomodguard goprintffuncname gosec gosmopolitan govet grouper iface importas inamedparam ineffassign interfacebloat intrange iotamixing loggercheck maintidx makezero mirror misspell mnd modernize musttag nakedret nestif nilerr nilnesserr nilnil noctx noinlineerr nolintlint nonamedreturns nosprintfhostport paralleltest perfsprint prealloc predeclared promlinter protogetter reassign recvcheck rowserrcheck sloglint spancheck sqlclosecheck staticcheck tagliatelle testableexamples testifylint testpackage thelper unconvert unparam unused usestdlibvars usetesting varnamelen wastedassign whitespace wrapcheck wsl_v5 zerologlint] 
INFO [loader] Go packages loading at mode 8767 (compiled_files|files|imports|types_sizes|deps|exports_file|name) took 594.149125ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 34.284458ms 
INFO [linters_context] importas settings found, but no aliases listed. List aliases under alias: key. 
INFO [linters_context/goanalysis] analyzers took 6m40.01852809s with top 10 stages: buildir: 1m4.263510518s, goimports: 26.674085168s, wastedassign: 24.427884127s, golines: 16.488936422s, buildssa: 14.70006813s, gosec: 13.133288248s, dupl: 10.259956876s, misspell: 6.773383795s, unparam: 5.947891207s, gci: 5.757882784s 
INFO [runner/exclusion_rules] Skipped 145 issues by rules: [Path: "_test.go", Linters: "errchkjson, maintidx, funlen, varnamelen, err113, errcheck"] 
INFO [runner/exclusion_rules] Skipped 1 issues by rules: [Path: "test_config.go", Linters: "mnd"] 
INFO [runner] Issues before processing: 9452, after processing: 2 
INFO [runner] Processors filtering stat (in/out): path_relativity: 9452/9452, generated_file_filter: 9452/221, exclusion_rules: 221/75, source_code: 2/2, path_prettifier: 2/2, filename_unadjuster: 9452/9452, exclusion_paths: 9452/9452, nolint_filter: 75/2, diff: 2/2, uniq_by_line: 2/2, max_per_file_from_linter: 2/2, severity-rules: 2/2, max_same_issues: 2/2, max_from_linter: 2/2, path_shortener: 2/2, cgo: 9452/9452, fixer: 2/2, sort_results: 2/2, path_absoluter: 9452/9452, invalid_issue: 9452/9452 
INFO [runner] processing took 31.059462ms with stages: generated_file_filter: 15.238333ms, path_relativity: 8.923751ms, nolint_filter: 5.148749ms, cgo: 710.626µs, invalid_issue: 659.5µs, exclusion_rules: 181.708µs, path_absoluter: 80.834µs, source_code: 60.583µs, filename_unadjuster: 41.084µs, sort_results: 7.792µs, max_same_issues: 2.125µs, uniq_by_line: 1.75µs, path_shortener: 750ns, exclusion_paths: 542ns, fixer: 501ns, max_from_linter: 375ns, diff: 126ns, max_per_file_from_linter: 125ns, path_prettifier: 124ns, severity-rules: 84ns 
INFO [runner] linters took 19.484956542s with stages: goanalysis_metalinter: 19.45379175s 
internal/app/api/orders/list.go:49:2: SA4006: this value of offset is never used (staticcheck)
        offset := query.Offset
        ^
internal/services/addresses/add_to_user.go:71:3: SA4006: this value of point is never used (staticcheck)
                point, err := s.geocodeAddress(ctx, input)
                ^
2 issues:
* staticcheck: 2
INFO File cache stats: 2 entries of total size 5.2KiB 
INFO Memory: 201 samples, avg is 1859.8MB, max is 3454.1MB 
INFO Execution took 20.130030583s 
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a [sponsor/backer](https://donate.golangci.org) of this project.

**Comment by ldez** [comment]
https://github.com/dominikh/go-tools/issues/1698

**Comment by nicholasngai** [comment]
Looks like https://github.com/dominikh/go-tools/issues/1698 is closed now! When will we see a release with the fix?

**Comment by ldez** [comment]
> Looks like https://github.com/dominikh/go-tools/issues/1698 is closed now!

Yes, a [PR](https://github.com/golangci/golangci-lint/pull/6367) is already opened on the golangci-lint repository.

> When will we see a release with the fix?

When it will be ready, and when you will pay for it.

https://donate.golangci.org

---

For people that put 👎 on my comment: I do my best every day for you (golangci-lint users).

Maybe you are thinking that this is just a dependency update, but it is not.

It's easy to demand, it's much harder to do what I do.

Why I continue to maintain golangci-lint if I constantly receive this kind of feedback?
Why I continue to maintain golangci-lint despite the fact I don't have enough money to live?

I don't know, but be sure that I'm a human and all these things have an impact on me.

You can continue to close your eyes about what OSS maintainers are living, but the reality will still exist, sorry if you don't want to see it.
